guides/pull-requests.md

```diff
@@ -4,52 +4,109 @@
+other PRs merged before the release will then use this version number, unless
+one of those PRs introduces a change that requires a more significant bump.
 
-For example, if the first PR introduced a bugfix, the version number would increase to <code>\_.\_.**X**</code>, but if a subsequent PR introduced new features, the version number would need to increase again to <code>\_.**X**.\_</code>
+For example, if the first PR introduced a bugfix, the version number would
+increase to <code>\_.\_.**X**</code>, but if a subsequent PR introduced new
+features, the version number would need to increase again to
+<code>\_.**X**.\_</code>
 
-### Status labels
+### Using Github PR Labels
 
-To help draw a user's attention to new features while navigating the docs, we use status labels in the sidebar navigation, and within the example's documentation.
+We use a variety of Labels to manage the status of a PR.
 
-Any affected examples in your PR should have an appropriate label, either `new`, `updated` or `deprecated`, added to the [sidebar navigation](/templates/_layouts/docs.html). This can by passing a third argument to any instance of `side_nav_item()` in that file, e.g.:
+When creating a PR, determine what types of review your PR requires and apply
+the appropriate Labels - for instance, `Review: Design needed` and
+`Review: Code needed` for a PR that requires design and dev reviews.
+
+When reviewing a PR, after performing the necessary reviews, remove the
+appropriate `needed` Labels and apply the relevant `+1` Labels instead,
+indicating that the PR has passed these specific review types. Alternatively,
+if your review indicates that further work is needed for the PR to pass review,
+apply the relevant `-1` Labels.
+
+### Submitting & merging a PR
+
+After the necessary review steps have been completed and the PR is ready to be
+merged, the creator of the PR should merge it themself. The type of merge to use
+should be decided using the following logic:
+
+- If the PR is small - for example, a PR consisting of 2 commits that modify
+  fewer than a dozen lines of code - `Squash and merge` should be used
+- If the PR is medium or large - for example, a PR consisting of 10 commits that
+  modify 100 lines of code - `Rebase and merge` should be used
+
+Ultimately, use your best judgment; if you feel like your PR is arguably too
+large to comfortably `Squash and merge`, then default to `Rebase and merge`.
+
+## Documentation site status labels
+
+To help draw a user's attention to new features while navigating the docs, we
+use status labels in the sidebar navigation, and within the example's
+documentation.
+
+Any affected examples in your PR should have an appropriate label - `new`,
+`updated`, or `deprecated` - added to the
+[sidebar navigation](/templates/_layouts/docs.html). This can be done by passing
+a third argument to any instance of `side_nav_item()` in that file, e.g.:
```

Comment:
I believe this has actually been refactored into using a macro. 

https://github.com/search?q=repo%3Acanonical%2Fvanilla-framework%20side_nav_item(&type=code

We no longer manually call `side_nav_item`; it look like this was updated in #4681 but was never added to the README. So we can probably remove this bit; we pass in the change status from `releases.yaml` instead.

Reply:
Yes, when you change the status of existing navigation item it's taken from `releases.yaml`. 
Only when you add new page you have to add it to `side-navigation.yaml`. So no need to manually mess with templates anymore.

It was changed a while ago, but seems we forgot to update this doc.